src/editors/containers/TextEditor/components/ImageSettingsModal/hooks.test.js

```diff
@@ -0,0 +1,314 @@
+          locked: null,
+        })).toEqual({ ...reducedDims });
+      });
+      it('returns getValidDimensions if locked and local has changed', () => {
+        const getValidDimensions = (args) => ({ getValidDimensions: args });
+        jest.spyOn(hooks, hookKeys.getValidDimensions).mockImplementationOnce(getValidDimensions);
+        const args = {
+          dimensions: { ...simpleDims },
+          local: { ...multiDims },
+          locked: { ...reducedDims },
+        };
+        expect(hooks.newDimensions(args)).toEqual(getValidDimensions(args));
+      });
+    });
+    describe('lockDimensions', () => {
+      it('does not call setLocked if lockInitialized is false', () => {
+        setState.locked = jest.fn();
+        hooks.lockDimensions({
+          dimensions: simpleDims,
+          setLocked: setState.locked,
+          lockInitialized: false,
+        });
+        expect(setState.locked).not.toHaveBeenCalled();
+      });
+      it(
+        'calls setLocked with the given dimensions and minInc, including gcd',
+        () => {
+          setState.locked = jest.fn();
+          hooks.lockDimensions({
+            dimensions: simpleDims,
+            setLocked: setState.locked,
+            lockInitialized: true,
+          });
+          expect(setState.locked).toHaveBeenCalledWith({
+            ...simpleDims,
+            minInc: { gcd: 1, ...simpleDims },
+          });
+          setState.locked.mockClear();
+
+          hooks.lockDimensions({
+            dimensions: multiDims,
+            setLocked: setState.locked,
+            lockInitialized: true,
+          });
+          expect(hooks.findGcd(multiDims.width, multiDims.height)).toEqual(7);
+          expect(setState.locked).toHaveBeenCalledWith({
+            ...multiDims,
+            minInc: { gcd, ...reducedDims },
+          });
+        },
+      );
+    });
+    describe('dimensionLockHooks', () => {
+      beforeEach(() => {
+        mockState();
+        jest.spyOn(hooks, hookKeys.lockDimensions).mockImplementationOnce(
+          ({ dimensions }) => setState.locked({ ...dimensions }),
+        );
+        hook = hooks.dimensionLockHooks({ dimensions: simpleDims });
+      });
+      afterEach(() => {
+        restoreState();
+      });
+
+      test('locked is initially null', () => {
+        expect(hook.locked).toEqual(null);
+      });
+      test('initializeLock calls setLockInitialized with true', () => {
+        hook.initializeLock();
+        expect(setState.lockInitialized).toHaveBeenCalledWith(true);
+      });
+      test('lock is successful if lockInitialized is true', () => {
+        mockStateVal(stateKeys.lockInitialized, true, setState.lockInitialized);
+        hook = hooks.dimensionLockHooks({ dimensions: simpleDims });
+        expect(hook.locked).toEqual(null);
+        hook.lock();
+        expect(setState.locked).toHaveBeenCalledWith({ ...simpleDims });
+      });
+      test('unlock sets locked to null', () => {
```

Comment:
Any reason we're doing locked: true unlocked: null instead of locked: true unlocked: false. Seems we're not bound to that convention, and it seems a little unintuitive?

Reply:
because locked is an object, and I want to test against truthiness not actual "===false".

Reply:
That makes sense.